src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -658,6 +658,36 @@ public ProjectState UpdateAdditionalDocument(TextDocumentState newDocument, bool
                 latestDocumentTopLevelChangeVersion: dependentSemanticVersion);
         }
 
+        public ProjectState UpdateDocumentsOrder(IEnumerable<DocumentId> documentIds)
+        {
+            var orderedDocumentIds = ImmutableList.CreateBuilder<DocumentId>();
+
+            foreach(var documentId in documentIds)
+            {
+                if (this.ContainsDocument(documentId))
+                {
+                    orderedDocumentIds.Add(documentId);
+                }
+            }
+
+            var orderedDocumentIdSet = new HashSet<DocumentId>(orderedDocumentIds);
+            var remainingDocumentIds = new List<DocumentId>();
+
+            foreach(var documentId in _documentIds)
+            {
+                if (!orderedDocumentIdSet.Contains(documentId))
+                {
+                    remainingDocumentIds.Add(documentId);
+                }
```

Comment:
i'm not sure i lke this at all.  why not require the ordered docs to contain all the files from the project?  this behavior of ordering what is iven and just tossing the rest at the end seems very hinky and it feels like it will mask bugs elsewhere.

Reply:
I'm hesitate to throw an error based on that logic, but maybe it makes sense so it ensures the caller is including everything, but I don't know for sure.

Reply:
I think I'm in agreement with you. I think I would rather throw an error to be absolutely sure when the method passes it is fully correct in relation to the order. I will make the adjustments